Dockerfile-cni-plugin

```diff
@@ -0,0 +1,40 @@
+# syntax=docker/dockerfile:1.4
+
+ARG BUILDPLATFORM=linux/amd64
+
+# Precompile key slow-to-build dependencies
+FROM --platform=$BUILDPLATFORM golang:1.18-alpine as golang
+WORKDIR /linkerd-build
+COPY --link go.mod go.sum .
+COPY --link ./cni-plugin ./cni-plugin
+COPY --link ./proxy-init ./proxy-init
+COPY --link ./internal ./internal
+RUN go mod download
+ARG TARGETARCH
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=$TARGETARCH GO111MODULE=on \
+    go build -o /go/bin/linkerd-cni -mod=readonly -ldflags "-s -w" -v ./cni-plugin/
+
+##
+## Runtime
+##
+
+FROM debian:bullseye-slim
+WORKDIR /linkerd
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    iptables \
+    inotify-tools \
+    procps \
+    jq && \
+    rm -rf /var/lib/apt/lists/*
+
+# We still rely on old iptables-legacy syntax.
+RUN update-alternatives --set iptables /usr/sbin/iptables-legacy \
+    && update-alternatives --set ip6tables /usr/sbin/ip6tables-legacy
+
+COPY --from=golang /go/bin/linkerd-cni /opt/cni/bin/
+COPY LICENSE .
+COPY cni-plugin/deployment/scripts/install-cni.sh .
```

Comment:
do we need to specify:

```
--from=go
```

in these copies so we can copy from the Go container? Or does this take the files from the build context? Bit of a noob question on my end.

Reply:
I'm pretty sure this works but I went ahead and changed it to `--from=go` to ensure we're doing it uniformly and maybe it saves us a few megabytes in the image size.

Reply:
Seems like the change is now breaking the docker build. Perhaps we can remove the `--from` for now. No idea why it can't compute the cache key when `--from` is specified. Hm